mysticeti-core/src/validator.rs

```diff
@@ -49,21 +51,31 @@ impl Validator {
         let mut binding_network_address = network_address;
         binding_network_address.set_ip(IpAddr::V4(Ipv4Addr::UNSPECIFIED));
 
-        let metrics_address = parameters
-            .metrics_address(authority)
-            .ok_or(eyre!("No metrics address for authority {authority}"))
-            .wrap_err("Unknown authority")?;
-        let mut binding_metrics_address = metrics_address;
-        binding_metrics_address.set_ip(IpAddr::V4(Ipv4Addr::UNSPECIFIED));
+        // Boot the prometheus server only when a registry is not passed in. If a registry is passed in
+        // we assume that an upstream component is responsible for exposing the metrics.
+        let (registry, metrics_handle) = if let Some(registry) = registry {
+            (registry, Handle::current().spawn(async { Ok(()) }))
```

Comment:
not proud of spawning an empty future, but will avoid touching the handler join for now - trying to do minimal refactoring until it is really needed.

Reply:
Yeah, I think this is ok for now